enterprise/coderd/templates_test.go

```diff
@@ -65,6 +68,25 @@ func TestTemplates(t *testing.T) {
 		assert.True(t, updated.Deprecated)
 		assert.NotEmpty(t, updated.DeprecationMessage)
 
+		notifs := []*testutil.Notification{}
+		for _, notif := range notifyEnq.Sent {
+			if notif.TemplateID == notifications.TemplateTemplateDeprecated {
+				notifs = append(notifs, notif)
+			}
+		}
+		require.Equal(t, 2, len(notifs))
+
+		expectedSentTo := []string{user.UserID.String(), anotherUser.ID.String()}
+		slices.Sort(expectedSentTo)
+
+		sentTo := []string{}
+		for _, notif := range notifs {
+			sentTo = append(sentTo, notif.UserID.String())
+		}
+		slices.Sort(sentTo)
+
+		assert.Equal(t, expectedSentTo, sentTo)
+
```

Comment:
suggestion: validate that no notification is sent to users of a different template